web/src/helpers/hooks/useIdeState.js

```diff
@@ -64,61 +98,71 @@ export const useIdeState = () => {
             : payload.message,
           isLoading: false,
         }
-      case 'setIdeType':
+      case 'setLayout':
         return {
           ...state,
-          ideType: payload.message,
+          layout: payload.message,
         }
-      case 'setLayout':
+      case 'updateCamera':
         return {
           ...state,
-          layout: payload.message,
+          camera: payload.camera,
+        }
+      case 'updateViewerSize':
+        return {
+          ...state,
+          viewerSize: payload.viewerSize,
         }
       case 'setLoading':
         return {
           ...state,
           isLoading: true,
         }
+      case 'resetLoading':
+        return {
+          ...state,
+          isLoading: false,
+        }
       default:
         return state
     }
   }
 
-  function dispatchMiddleware(dispatch, state) {
-    return ({ type, payload }) => {
-      switch (type) {
-        case 'render':
-          cadPackages[state.ideType]
-            .render({
-              code: payload.code,
-              settings: {
-                camera: payload.camera,
-                viewerSize: payload.viewerSize,
-              },
-            })
-            .then(({ objectData, message, status }) => {
-              if (status === 'insufficient-preview-info') return
-              if (status === 'error') {
-                dispatch({
-                  type: 'errorRender',
-                  payload: { message },
-                })
-              } else {
-                dispatch({
-                  type: 'healthyRender',
-                  payload: { objectData, message },
-                })
-              }
-            })
-          dispatch({ type: 'setLoading' })
-          break
-
-        default:
-          return dispatch({ type, payload })
-      }
-    }
-  }
-
   const [state, dispatch] = useReducer(reducer, initialState)
-  return [state, dispatchMiddleware(dispatch, state)]
+  mutableState = state
```

Comment:
This is the one hacky thing I did in regards to the thunks. I'm okay with it since now all the new thunks can use the getState function() and shouldn't need to implement more hacks.